dali/kernels/slice/slice_kernel_utils.h

```diff
@@ -104,6 +104,31 @@ inline std::tuple<int64_t, int64_t, int64_t> CalcPadCopyExtents(int64_t anchor,
   return std::tuple<int64_t, int64_t, int64_t>{pad_before, to_copy, pad_after};
 }
 
+
+template <typename OutputType, typename InputType, int Dims>
+bool CanRunPlainCopy(const TensorShape<Dims> &out_strides,
+                     const TensorShape<Dims> &in_strides,
+                     const TensorShape<Dims> &out_shape,
+                     const TensorShape<Dims> &in_shape,
+                     const SliceArgs<OutputType, Dims> &args) {
+  // if there's type conversion we can't run memcpy
+  if (!std::is_same<OutputType, InputType>::value)
+    return false;
+
+  auto default_out_strides = GetStrides(out_shape);
+  auto default_in_strides = GetStrides(in_shape);
+
+  // If the strides are not the default ones, or the window anchor and shape
+  // are different than the bounds of the input, we can't run plain memcpy
+  for (int d = 0; d < Dims; d++) {
```

Comment:
Isn't there a special case for 1D data that you could use memcpy even when the anchor and shape are different than the input bounds, but assuming the default stride?

Reply:
I mean, it still would be a contiguous chunk of memory.

Reply:
Yeah, I thought about that, but decided not to over-engineer it too much